claf/learn/experiment.py

```diff
@@ -63,13 +65,17 @@ def common_setting(self, mode, config):
 
         cuda_devices = self._get_cuda_devices()
         self.config.cuda_devices = cuda_devices
+        self.config.slack_url = getattr(self.config, "slack_url", False)
```

Comment:
`getattr(self.config, "slack_url", None)` would be better, because this makes L324 `if self.config.slack_url is not None` be evaluated as `True` when `slack_url` does not exist in `config`. Such a case can occur when options are loaded from a base config file in which `slack_url` does not exist.

Another option would be to change L324 to `if not self.config.slack_url:`. In fact, L128 also uses `if self.config.slack_url:`.

Also, it seems like `slack` flag in some of the files inside `base_config` is deprecated; it would be better if we remove them.

Reply:
Thanks for your feedback :)

I think that maintain the ```getattr(self.config, "slack_url", False)``` part and change ```if not self.config.slack_url:``` -> ```if self.config.slack_url:``` (L324) is looks better.
What do you think of it?

I'll remove all `slack` flags in `base_config`.

Reply:
Oh, I meant to suggest to change L324 to `if self.config.slack_url`. Putting `not` there was a mistake. 😂 Agreed :)

Reply:
Resolved by commit https://github.com/naver/claf/commit/518b1726f355ae2e61d42898a8cf5c7960e14d51